src/source-control/GitSourceControl.ts

```diff
@@ -1,9 +1,31 @@
-import type { Commit, CommitHash } from "../shared/types";
+import { func } from "prop-types";
+import type { BranchName, Commit, CommitHash } from "../shared/types";
 import type {
   SourceControl,
   SourceControlRepositoryUpdateListener,
 } from "./SourceControl";
 
+var randomWords = require("random-words");
+let localRefPrefix: string = "refs/heads/";
+
+function refIsLocal(refString: string): boolean {
+  return refString.startsWith(localRefPrefix);
+}
+
+function localRefToBranchName(refString: string): string {
+  if (!refIsLocal(refString)) return refString;
+  return refString.substr(localRefPrefix.length);
+}
+
+function isSttackBranch(branch: BranchName): boolean {
+  if (branch.startsWith("sttack-")) return true;
+  return false;
+}
+
+function createSttackBranch(commit: Commit): string {
+  let branch: BranchName = `sttack-${randomWords({ exactly: 4, join: "-" })}`;
+  return branch;
```

Comment:
nit, if TS can handle `BranchName` as the return type:

```suggestion
function createSttackBranch(commit: Commit): BranchName {
  return `sttack-${randomWords({ exactly: 4, join: "-" })}`;
```